components/request_otr/browser/request_otr_navigation_throttle.cc

```diff
@@ -52,7 +53,10 @@ RequestOTRNavigationThrottle::MaybeCreateThrottleFor(
           net::features::kBraveFirstPartyEphemeralStorage)) {
     return nullptr;
   }
-  DCHECK(ephemeral_storage_service);
+  DCHECK(ephemeral_storage_service ||
+         profile_metrics::GetBrowserProfileType(
```

Comment:
but the code below assumes `ephemeral_storage_service` is really non-null, so we shouldn't proceed with a non-existing service
Perhaps we should just return if this is not the profile type we want?

Reply:
I am guessing we only want to proceed if it's a `kRegular` profile type?

Reply:
@iefremov, PTAL https://github.com/brave/brave-core/pull/21134/commits/55a9f20ad5c6a1b2e7affb66dc9af89d17acf395

Reply:
let's just exclude the system one, i couldn't infer what we need for incognito from the [spec](https://docs.google.com/document/d/1J1x-pNyT5CENzdb-RxzPxLxFLE4c_PRlz6T-usJUQR4/edit), and Mark is no longer with us